dns-query/src/index.ts

```diff
@@ -4,4 +4,10 @@ import { makeConfig } from './config'
 
 const NAME = 'DNS-Query'
 
-export = { NAME, makeConfig, makeExecute, ...expose(util.wrapExecute(makeExecute())) }
+const execute = makeExecute()
+export = {
+  NAME,
+  makeConfig,
+  makeExecute,
+  ...expose(util.wrapExecute(execute.call), '', execute.config),
```

Comment:
I was thinking of just send the function here as before, and the middleware reads config from it directly.

```suggestion
  ...expose(util.wrapExecute(execute)),
```